src/dune

```diff
@@ -0,0 +1,12 @@
+(library
```

Comment:
this should not be a library, since it contains side-effects IMO. it's rendering to the DOM at the toplevel. This should be an entry in `melange.emit`.

Reply:
Assuming I understand the reasoning behind this side-effects rule (I still don't 😄 ), doing what you request involves either moving the React component module up to the root folder so it can become an `entry`, or have the `melange.emit` moved to `src`. 

Both are not desirable for the following reasons:
- moving sources to root folder doesn't scale, as you add more entry points you end up with root folder filled with React components
- having `melange.emit` in inner folder doesn't scale either, because either it won't be able to "see" sources outside of it, or you will end up with multiple `melange.emit` stanzas, which leads to duplication of js artifacts of any library that is shared across these emits, slowing down webpack, removing reuse across JS bundles, etc.

Reply:
ref: https://github.com/ocaml/dune/issues/7017